### DISCUSSION TITLE
Labels specific user-features

### DISCUSSION THREAD

**Discussion by LouisLouiseKay** [funmusicplace/mirlo#157] [Ideas]
When a user identifies as a music-label, they should have access to specific features : 

- [x] Profile picture displays roster of managed artists
- [x] When a label-user "creates an artist" they should be able to type the name of an artist and request access to this artist account management.
- [x] If an artist has only made "a few releases" under a certain label, the artist should be able to precise during the album creation/editing process that "this album was made with xxxx label".
- [ ] When an artist has 3 albums in their account under a label, the label can only see and edit those 3 albums when looking at the artist profile editing page.

To be determined if Labels get full access to artist features, or just to some of them. I'll investigate how bandcamp does it because it's pretty well integrated.

**Comment by simonv3** [comment]
For the sharing of album management with another (label) account I was thinking of introducing two new fields in the TrackGroup table:

## Step 1: changes to current code

* [x] Change the API so that it doesn't reference the user ID of the artist being managed. `/user/1/artist/1` becomes `/manage/artist/1`. This just simplifies some things around logic who can and who can't manage artists. This will require some pretty significant code changes across the app around permission checks. I think most of those already have tests written so that's good.

## Step 2: admin can edit artists and trackgroups

* [x] Implement functionality for an admin to edit artists or trackgroups. This should be possible already but is mainly being blocked by logic caused by the complexity of step 1. So once that's implemented we have to test whether this works.

## Step 3: transfer artist ownership

* [ ] Add a "transferOwnership" endpoint at `/manage/artist/:id/transferOwnership` that moves an artist to a new `userId`.
* [ ] Add a button on the artist page next to the `Delete` button that lets an artist do this with a pop up where they enter a userId. The pop up should also explain that this means that the income generated by the artist will go to the new user account. 
* [ ] **Question**: do we need the receiving user to confirm the transfer? In which case this is a bit more complicated. 

## Step 3: Other users can manage artists

* [ ] Implement a `userCanManageArtist` table
* [ ] Add logic to the permissions logic for artist editing that checks this table
* [ ] Add a differentiator between "owned" artists "managed" artists to a user's profile page. Possible in a new Artists tab?
* [ ] Add managed artists to the dropdown menu.
* [ ] Figure out any spots in the UI where this might now be broken? 

## Step 4: Say a different user can receive payouts for an album purchase

* [x] add `ownerId` to `TrackGroup`
* [x] When determining where payout store goes, first check `trackGroup.ownerId` and then `trackGroup.artist.userId`. Redirect to appropriate Stripe
* [x] On receiving payment confirmation, send email to `trackGroup.ownerId` before `trackGroup.artist.userId`

## Step 4: Other users can manage trackGroups

| managedByUserId | trackGroupId |
|---|---|

The first one will be who can "edit" an album.

I have some questions here around who should be allowed to edit an album, and if it could exclude the user who owns the artist account (for ex, a previous label owns the rights of the trackgroup, so the artist shouldn't be able to edit it). That could be TBD for later though

**Reply by LouisLouiseKay** [reply]
We can absolutely do 3 later. I think ultimately it's not critical wether we go with 1 or 2 because in either cases tensions could arise (because tensions do arise between a label and an artist in real life). We can leave it to "either" for now, see if people complain about that. If the difference in complexity is substantial with just having one manageUser we can even go with (1) but I think (2) is more agnostic and "sort this out yourself".

**Comment by timglorioso** [comment]
Whenever I think about labels this is where my brain starts to go: contextual navigation

![navigation_context](https://github.com/user-attachments/assets/b7873019-13a6-4fd6-8d66-9382311a2395)

**Reply by timglorioso** [reply]
And no I did not use AI to generate this sloppy Inkscape drawing :P

**Comment by timglorioso** [comment]
I've also considered calling them Groups and letting the user choose lingo (such as label, collective, etc.). Trouble then is needing translation, which if I'm being honest feels like a Platform Problem.

**Reply by simonv3** [reply]
I also want to call them groups (I've been saying collectives) but there's the issue of it being identifiable to labels. I think if someone chooses to name their thing a "collective" then we can call it that in the translation too (ie, the string would be "View this **collective**", "bekijk deze **collective**"). 

What do you mean by Platform Problem?

**Reply by timglorioso** [reply]
Maybe it doesn't even need a name like that. What if we asked for the group name and used that instead of the generic "Label"? I think it's more effective to let the real world speak for itself and feel like part of the system.

tl;dr I was wrong about Platform Problem.

I was thinking of things like handling lots of traffic, things that become problems at scale. Translation felt like that to me in the sense that if you #beyourownplatform you're not having to translate the manage UI because it's only for you. But really that only applies in the rare case that you're writing HTML directly. Any CMS will have UX copy of some kind.

**Reply by timglorioso** [reply]
I guess HTML is always in English too, but at that point you're working with symbols

**Comment by timglorioso** [comment]
One more thread on this discussion for now. I at least want to put this on paper. I imagine this is a future consideration.

This is the model I think would be most flexible. An Account can be associated with any number of Groups and Artists, and always includes a personal profile. People can still use multiple Accounts if they want. I'm not sure if Artists should be able to belong to multiple Groups at the same time. I don't see why not, but it doesn't seem like a common scenario.

```mermaid
erDiagram
    Account }|--o{ Artist : has
    Artist }|--o| Group : "belongs to"
    Account }|--o{ Group : "belongs to"
```

**Comment by simonv3** [comment]
I'm going to close this. Label pages are implemented and other stuff should be handled as more specific feature changes.
